pkg/kudoctl/util/kudo/kudo_test.go

```diff
@@ -574,3 +574,56 @@ func TestKudoClient_UpdateOperatorVersion(t *testing.T) {
 		}
 	}
 }
+
+func TestKudoClient_DeleteInstance(t *testing.T) {
+	testInstance := v1alpha1.Instance{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "kudo.dev/v1alpha1",
+			Kind:       "Instance",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"controller-tools.k8s.io": "1.0",
+			},
+			Name: "test",
+		},
+		Spec: v1alpha1.InstanceSpec{
+			OperatorVersion: v1.ObjectReference{
+				Name: "test-1.0",
+			},
+		},
+	}
+
+	installNamespace := "default"
+
+	k2o := newTestSimpleK2o()
+
+	_, err := k2o.clientset.KudoV1alpha1().Instances(installNamespace).Create(&testInstance)
+	if err != nil {
+		t.Fatalf("error creating instance in tests setup for")
+	}
+
+	err = k2o.DeleteInstance("nonexisting-instance-name", installNamespace)
```

Comment:
I think we prefer table tests for these kinds of tests. I would probably prefer to stick with that

Reply:
I'll look into that. IIRC, I started this as a table test, but the code got quite convoluted to handle the various test results.